file/file.go

```diff
@@ -0,0 +1,70 @@
+// Package file - Handle file operations
+package file
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+)
+
+const filename = "tasks.json"
+
+// File - file generic struct. T generic.
+type File[T any] struct{}
+
+// New - instantiates a File struct.
+func New[T any]() *File[T] {
+	return &File[T]{}
+}
+
+// Write - writes tasks into a JSON file
+func (f *File[T]) Write(data *[]T) error {
+	asJSON, err := json.Marshal(data)
+	if err != nil {
+		return err
+	}
+
+	err = os.WriteFile(filename, asJSON, 0644)
+	if err != nil {
+		return err
```

Comment:
You have two cases in the function where you can return error, so you should wrap them both with some context, so that the user better will understand what went wrong: marshaling or writing file.